trade-finance-manager-ui/scripts/enable-find-utilisation-reports-by-year-dropdown-autocomplete.ts

```diff
@@ -0,0 +1,29 @@
+const enableFindUtilisationReportsByYearDropdownAutocomplete = (): void => {
+  const findUtilisationReportsByYearForm: HTMLFormElement | null = document.querySelector('form#find-utilisation-reports-by-year--form');
+
+  if (!findUtilisationReportsByYearForm) {
+    return;
+  }
+
+  const yearInput: HTMLInputElement | null = document.querySelector('input#find-utilisation-reports-by-year--year-input');
+  if (!yearInput) {
+    return;
+  }
+
+  const bankRadios: NodeListOf<HTMLInputElement> = findUtilisationReportsByYearForm.querySelectorAll('input.govuk-radios__input');
+
+  bankRadios.forEach((bankRadio) => {
+    const bankId = bankRadio.value;
+    const dataListId = `datalist--bankId-${bankId}`;
+
+    if (bankRadio.checked) {
+      yearInput.setAttribute('list', dataListId);
```

Comment:
Is `list ` custom data attribute, if yes then it should be `data-list` please https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes.

Reply:
It is not a custom data attribute. It is an attribute on `input` elements which allow it to link up with a `datalist` tag (this is builtin HTML 5 functionality)